app/src/main/java/com/example/android/uamp/fragments/MediaItemListScreen.kt

```diff
@@ -0,0 +1,127 @@
+import coil.compose.AsyncImage
+import com.example.android.uamp.MediaItemData
+import com.example.android.uamp.R
+import com.example.android.uamp.viewmodels.MainActivityViewModel
+import com.example.android.uamp.viewmodels.MediaItemFragmentViewModel
+
+/**
+ * This particular instance of MediaItemDescription serves as the function which keeps track of the media items state
+ * and recomposes the UI upon any changes
+ *
+ * @param mediaItemFragmentViewModel to reference MediaItemFragmentViewModel functions
+ * @param mainActivityViewModel to reference MainActivityViewModel functions
+ */
+@Composable
+fun MediaItemDescription(
+    mediaItemFragmentViewModel: MediaItemFragmentViewModel,
+    mainActivityViewModel: MainActivityViewModel
+) {
+    val currentMediaItems by mediaItemFragmentViewModel.mediaItems.observeAsState()
+    currentMediaItems?.let { mediaItems ->
+        MediaItemDescription(mediaItems, mainActivityViewModel)
+    }
+}
+
+/**
+ * This particular instance of MediaItemDescription displays the mediaItems in a vertical list scroll format
+ *
+ * @param mediaItems List of mediaItems in the selected branch
+ * @param mainActivityViewModel to reference MainActivityViewModel functions
+ */
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+private fun MediaItemDescription(
+    mediaItems: List<MediaItemData>,
+    mainActivityViewModel: MainActivityViewModel
+) {
+    if (mediaItems.isEmpty())
+        Text("Media Items Empty")
+    else
+        LazyColumn(modifier = Modifier.fillMaxSize()) {
+            for (mediaItem in mediaItems) {
+                item { MediaItem(mediaItem, mainActivityViewModel) }
+            }
+        }
+}
+
+/**
+ * This function describes the UI of a mediaItem instance
+ *
+ * @param mediaItemData to reference details of a mediaItem in the current branch
+ * @param mainActivityViewModel to reference MainActivityViewModel functions
+ */
+@Composable
+private fun MediaItem(mediaItemData: MediaItemData, mainActivityViewModel: MainActivityViewModel) {
+    val mediaItemArtDimen = dimensionResource(R.dimen.media_item_art)
+    val imageModel = mediaItemData.mediaItem.mediaMetadata.artworkUri
+    val mediaItemTitle = mediaItemData.title
+    val mediaItemArtist = mediaItemData.subtitle
+
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .clickable { mainActivityViewModel.mediaItemClicked(mediaItemData) },
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        AsyncImage(
+            model = imageModel,
+            contentDescription = null,
+            modifier = Modifier
+                .width(mediaItemArtDimen)
+                .height(mediaItemArtDimen),
+            placeholder = painterResource(id = R.drawable.default_art),
+            error = painterResource(id = R.drawable.default_art)
+        )
+        Column(horizontalAlignment = Alignment.CenterHorizontally) {
+            Text(
+                text = mediaItemTitle, modifier = Modifier
```

Comment:
You may want to run the code formatter on this (and other files you changed). There should be a newline after "text = mediaItemTitle,"

Reply:
The formatter doesn't seem to complain as long as the line does not surpass a certain margin on the screen. Making these changes manually.